README.md

```diff
@@ -132,3 +132,609 @@ gradleTestUtils {
+    - [Multi-Version Testing](#multi-version-testing)
+- [File Operations](#file-operations)
+    - [Working with Files](#working-with-files)
+    - [Build Gradle](#build-gradle)
+    - [Settings Gradle](#settings-gradle)
+    - [Plugin Management](#plugin-management)
+      - [Testing with External Plugins](#testing-with-external-plugins)
+    - [Gradle Files](#gradle-files)
+    - [Java Source Files](#java-source-files)
+    - [Properties Files](#properties-files)
+    - [YAML Files](#yaml-files)
+    - [Arbitrary Files](#arbitrary-files)
+    - [Directories](#directories)
+    - [Build Directory](#build-directory)
+- [Executing Gradle Builds](#executing-gradle-builds)
+    - [Basic Execution](#basic-execution)
+    - [Environment Variables](#environment-variables)
+- [Assertions](#assertions)
+    - [Fluent Assertion API](#fluent-assertion-api)
+    - [Task Outcome Assertions](#task-outcome-assertions)
+    - [Output Assertions](#output-assertions)
+    - [File Assertions](#file-assertions)
+- [Error Prone Checks](#error-prone-checks)
+    - [Why We Ship Error Prone Checks](#why-we-ship-error-prone-checks)
+    - [Configuring Error Prone](#configuring-error-prone)
+
+## Quick Start
+
+### IDE Setup - Syntax Highlighting
+
+Add the [gradle-idea-language-injector](https://github.com/palantir/gradle-idea-language-injector) plugin to your root `build.gradle`:
+
+```gradle
+plugins {
+    id 'com.palantir.idea-language-injector' version '<version>'
+}
+```
+
+This plugin automatically generates IntelliJ IDEA language injection configurations, enabling syntax highlighting for string literals.
+
+### Write your first test
+
+```java
+import com.palantir.gradle.testing.junit.GradlePluginTests;
+import com.palantir.gradle.testing.execution.GradleInvoker;
+import com.palantir.gradle.testing.project.RootProject;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import static com.palantir.gradle.testing.assertion.GradlePluginTestAssertions.assertThat;
+
+@GradlePluginTests
+class MyPluginTest {
+    @BeforeEach
+    void setup(RootProject project) {
+        project.buildGradle().plugins().add("my-plugin");
+        project.buildGradle().append("""
+                myPlugin {
+                    enabled = true
+                }
+                """);
+    }
+
+    @Test
+    void plugin_applies_successfully(GradleInvoker gradle, RootProject project) {
+        assertThat(gradle.withArgs("tasks").buildsSuccessfully())
+            .output()
+            .contains("myTask");
+    }
+}
+```
+
+## Core Concepts
+
+### The `@GradlePluginTests` Annotation
+
+Mark test classes with `@GradlePluginTests` to enable Gradle plugin testing. This automatically:
+- Creates isolated project directories for each test
+- Runs tests against multiple Gradle versions read from the `PluginTestingExtension` see [Multi-Version Testing](#multi-version-testing)
+- Provides parameter injection for test infrastructure
```

Comment:
I don't think this is useful detail to know how to write tests. It's useful to know what the annotation does, but in terms of , how do I write a test that works, it's buried below these facts